src/utils.rs

```diff
@@ -26,3 +27,36 @@ impl<T> Deref for Lazy<T> {
         self.cell.get_or_init(self.init)
     }
 }
+
+trait UpcastFrom<T: ?Sized> {
+    fn any_from(value: &T) -> &Self;
+    fn any_mut_from(value: &mut T) -> &mut Self;
+}
+
+pub trait AsAny<U: ?Sized> {
```

Comment:
Why the generic parameter `U` here, if we only ever use it as `AsAny<dyn Any>`?

Reply:
You can not bound by trait object though.

Reply:
I think my question is: Why not just
```rust
pub trait AsAny {
    fn as_any(&self) -> &dyn Any;
    fn as_any_mut(&mut self) -> &mut dyn Any;
}

impl<T: Any> AsAny for T {
    fn as_any(&self) -> &dyn Any {
        self
    }

    fn as_any_mut(&mut self) -> &mut dyn Any {
        self
    }
}
```

Reply:
Hm, indeed.